src/mmw/js/src/draw/views.js

```diff
@@ -231,6 +231,8 @@ var DrawWindow = Marionette.LayoutView.extend({
             model: this.model,
             resetDrawingState: resetDrawingState
         }));
+
+        this.listenTo(App.map, 'change:lat, change:lng', resetDrawingState);
```

Comment:
You're right, I think this is too wide-ranging. I think it makes sense to do this for both free-draw and boundary, especially if it helped with the UtfGrid caching issue, but since it doesn't I don't think we should keep it. I'm going to close this PR.